### ISSUE TITLE
/admin/shutdown_room endpoint gives a 500 when attempting to ban an already-banned room

### ISSUE THREAD

**Issue by neutralinsomniac** [matrix-org/synapse#4645]
Pretty self-explanatory. Traceback below:
```
2019-02-14 09:52:12,895 - synapse.rest.client.v1.admin - 482 - INFO - POST-4071- Shutting down room '!QtykxKocfZaZOUrTwp:matrix.org'
2019-02-14 09:52:12,910 - synapse.http.server - 112 - ERROR - POST-4071- Failed handle request via <function JsonResource._async_render at 0x7f2110e04400>: <XForwardedForRequest at 0x7f20cc4c10f0 method='POST' uri='/_matrix/client/r0/admin/shutdown_room/!QtykxKocfZaZOUrTwp:matrix.org?access_token=<redacted>' clientp
roto='HTTP/1.1' site=8008>: Traceback (most recent call last):
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 1475, in gotResult
    _inlineCallbacks(r, g, status)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
--- <exception caught here> ---
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/http/server.py", line 81, in wrapped_request_handler
    yield h(self, request)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/http/server.py", line 316, in _async_render
    callback_return = yield callback(request, **kwargs)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/rest/client/v1/admin.py", line 484, in on_POST
    yield self.store.block_room(room_id, requester_user_id)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/room.py", line 509, in block_room
    desc="block_room",
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/_base.py", line 501, in _simple_insert
    self._simple_insert_txn, table, values,
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/_base.py", line 393, in runInteraction
    *args, **kwargs
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/_base.py", line 442, in runWithConnection
    inner_func, *args, **kwargs
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/enterprise/adbapi.py", line 306, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/python/compat.py", line 464, in reraise
    raise exception.with_traceback(traceback)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/twisted/enterprise/adbapi.py", line 297, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/_base.py", line 438, in inner_func
    return func(conn, *args, **kwargs)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/_base.py", line 314, in _new_transaction
    r = func(txn, *args, **kwargs)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/_base.py", line 521, in _simple_insert_txn
    txn.execute(sql, vals)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/_base.py", line 104, in execute
    self._do_execute(self.txn.execute, sql, *args)
  File "/home/jeremy/.synapse/lib/python3.5/site-packages/synapse/storage/_base.py", line 134, in _do_execute
    sql, *args
psycopg2.IntegrityError: duplicate key value violates unique constraint "blocked_rooms_idx"
DETAIL:  Key (room_id)=(!QtykxKocfZaZOUrTwp:matrix.org) already exists.
2019-02-14 09:52:12,929 - synapse.access.http.8008 - 302 - INFO - POST-4071- 174.98.59.17 - 8008 - {@piroko:pintobyte.com} Processed request: 2.062sec/0.003sec (0.056sec, 0.020sec) (0.055sec/0.184sec/45) 67B 500 "POST /_matrix/client/r0/admin/shutdown_room/!QtykxKocfZaZOUrTwp:matrix.org?access_token=<redacted> HTTP/1.1" "Mozilla/5.0 (X11; Fedora; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36" [6 dbevts]
```

**Comment by dklimpel** [comment]
In my opinion it is fixed: https://github.com/matrix-org/synapse/blob/b76f53bb79fc9f6f60de90f28a8194a48764d146/synapse/handlers/room.py#L1243-L1246

**Comment by richvdh** [comment]
looks like this was fixed by #4904.

**Comment by Robert061** [comment]
# #7ukkkkk
